### PR TITLE
Remove `leancode_lint`'s dependency on `flutter`

### LINKED ISSUE #45: `leancode_lint` should not depend on Flutter
raised [here](https://github.com/leancodepl/flutter_corelibrary/pull/44#issuecomment-979041786)

@shilangyu @Albert221 @RobertOdrowazLNCD

### PR DESCRIPTION
**Pull request by bartekpacia**
fix #45

### CODE REVIEW

File: packages/leancode_lint/README.md [hunk 1]
```diff
@@ -27,6 +27,8 @@ dependencies:
 
 ## Usage
 
+### App
+
 Add `include: package:leancode_lint/analysis_options.yaml` to
 `analysis_options.yaml` in your project. You might want to exclude some files
 (e.g generated frezeed models) from analysis.
```
Comment (shilangyu) [inline packages/leancode_lint/README.md:34]:
```suggestion
(e.g generated freezed models) from analysis.
```

File: packages/leancode_lint/README.md [hunk 2]
```diff
@@ -37,8 +39,25 @@ include: package:leancode_lint/analysis_options.yaml
...
-    - "**/*.freezed.dart"
+    - '**/*.g.dart'
+    - '**/*.freezed.dart'
+```
+
+### Package
+
+Add `include: package:leancode_lint/analysis_options_package.yaml` to
+`analysis_options.yaml` in your project. It includes additional lints for
+packages. You might want to exclude some files (e.g generated frezeed models)
```
Comment (shilangyu) [inline packages/leancode_lint/README.md:50]:
```suggestion
packages. You might want to exclude some files (e.g generated freezed models)
```

**Review by shilangyu — APPROVED** [review]

**Comment by Albert221** [comment]
What's the `.prettierrc` file for?

**Comment by bartekpacia** [comment]
@Albert221 It's here to solve the `"` vs `'` problem in `pubspec.yaml`. I remember @RobertOdrowazLNCD suggesting using Prettier for this on the last mobile weekly.

**Review by Albert221 — APPROVED** [review]
LGTM

**Pull request merged**